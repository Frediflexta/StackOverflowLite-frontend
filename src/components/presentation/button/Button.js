import React from 'react';
import propTypes from 'prop-types';

const Button = (props) => {
  const {
    className, type, text,
  } = props;

  return (
    <button
      type={type}
      className={className}
    >
      {text}
    </button>
  );
};

Button.propTypes = {
  className: propTypes.string.isRequired,
  type: propTypes.string,
  text: propTypes.string,
};

Button.defaultProps = {
  className: 'form-button',
  type: 'button',
  text: 'button'
};

export default Button;
