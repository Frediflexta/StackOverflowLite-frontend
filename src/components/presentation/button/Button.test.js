import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('<Button /> shallow rendering tests', () => {
  const wrapper = shallow(
    <Button
      className="test"
      type="button"
      text="Smurf"
    />
  );

  it('should render Button with props successfully', () => {
    expect(wrapper.find('.test').exists()).toBe(true);
    expect(wrapper.find('button').text()).toBe('Smurf');
  });

  it('matches the snapshot', () => {
    const tree = shallow(<Button />);
    expect(tree).toMatchSnapshot();
  });
});
