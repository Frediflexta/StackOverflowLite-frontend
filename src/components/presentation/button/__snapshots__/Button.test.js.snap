// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> shallow rendering tests matches the snapshot 1`] = `
<button
  className="form-button"
  type="button"
>
  button
</button>
`;
